lib/include.js

```diff
@@ -175,6 +175,83 @@ Inclusion.include = function (objects, include, options, cb) {
     cb && cb(err, objects);
   });
 
+  // Find the limit of items for `inq`
+  var inqLimit = 256;
+  if(self.dataSource && self.dataSource.settings &&
+    self.dataSource.settings.inqLimit) {
+    inqLimit = self.dataSource.settings.inqLimit;
+  }
+  /**
+   * Find related items with an array of foreign keys by page
+   * @param model The model class
+   * @param filter The query filter
+   * @param fkName The name of the foreign key property
+   * @param pageSize The size of page
+   * @param options Options
+   * @param cb
+   */
+  function findWithForeignKeysByPage(model, filter, fkName, pageSize, options, cb) {
+    var foreignKeys = [];
+    if (filter.where[fkName]) {
+      foreignKeys = filter.where[fkName].inq;
+    } else if (filter.where.and) {
+      // The inq can be embedded inside 'and: []'
+      for (var j in filter.where.and) {
```

Comment:
No `or: []` is needed as `include` only use `and`. We only deal with the generated `inq` for `include`.